src/link/MachO.zig

```diff
@@ -491,7 +491,7 @@ fn resolveSearchDir(
 ) !?[]const u8 {
     var candidates = std.ArrayList([]const u8).init(arena);
 
-    if (fs.path.isAbsolute(dir)) {
+    if (!fs.path.isAbsolute(dir)) {
```

Comment:
I was thinking of something along those lines (please feel free to do whatever you see fit with the diff):

```diff
diff --git a/src/link/MachO.zig b/src/link/MachO.zig
index 12556b53d..684f308a8 100644
--- a/src/link/MachO.zig
+++ b/src/link/MachO.zig
@@ -493,7 +493,17 @@ fn resolveSearchDir(
 
     if (fs.path.isAbsolute(dir)) {
         if (syslibroot) |root| {
-            const full_path = try fs.path.join(arena, &[_][]const u8{ root, dir });
+            const common_dir = if (std.Target.current.os.tag == .windows) blk: {
+                // We need to check for disk designator and strip it out from dir path so
+                // that we can concat dir with syslibroot.
+                // TODO we should backport this mechanism to 'MachO.Dylib.parseDependentLibs()'
+                const disk_designator = fs.path.diskDesignatorWindows(dir);
+                if (mem.indexOf(u8, dir, disk_designator)) |where| {
+                    break :blk dir[where..];
+                }
+                break :blk dir;
+            } else dir;
+            const full_path = try fs.path.join(arena, &[_][]const u8{ root, common_dir });
             try candidates.append(full_path);
         }
     }

```